packages/hardhat-core/src/internal/core/providers/http.ts

```diff
@@ -154,7 +166,7 @@ export class HttpProvider extends EventEmitter implements EIP1193Provider {
       if (this._isRateLimitResponse(response)) {
         // Consume the response stream and discard its result
         // See: https://github.com/node-fetch/node-fetch/issues/83
```

Comment:
Great catch. I'm trusting the comment, that the `discard` was a workaround for that `node-fetch` issue, and since we're not using `node-fetch` anymore I just removed this, in https://github.com/nomiclabs/hardhat/pull/2346/commits/6736e55b2c44a75a16ebad99955adb42faeb2616

Reply:
Addressed in [4bb5183](https://github.com/nomiclabs/hardhat/pull/2346/commits/4bb518399b3a9cd209acfa92600d63295fbea3e0)

Reply:
Thank you for pushing on this. You're right: [the undici docs for `request()`](https://github.com/nodejs/undici/blob/main/docs/api/Dispatcher.md#dispatcherrequestoptions-callback) say "All response bodies must always be fully consumed or destroyed." Addressed in https://github.com/nomiclabs/hardhat/pull/2346/commits/4bb518399b3a9cd209acfa92600d63295fbea3e0